src/client.rs

```diff
@@ -98,33 +104,18 @@ impl Tidal {
     }
 
     fn session_id(&self) -> String {
-        let credentials = self.credentials.clone();
-        let session_id = match &credentials.session_info {
-            None => panic!("A session needs to be obtained before using Tidal"),
-            Some(session_info) => match &session_info.session_id {
-                Some(session_id) => session_id,
-                None => panic!("An active sessions needs to be obtained before using Tidal"),
-            },
-        };
-        session_id.to_owned()
+        // Here it's safe to use unwrap because in ::new() we already checked that there's a valid session_info
+        self.credentials.session_info.as_ref().unwrap().session_id.as_ref().unwrap().to_owned()
     }
 
     fn country_code(&self) -> String {
-        self.credentials
-            .session_info
-            .as_ref()
-            .unwrap()
-            .country_code
-            .to_owned()
+        // Here it's safe to use unwrap because in ::new() we already checked that there's a valid session_info
+        self.credentials.session_info.as_ref().unwrap().country_code.to_owned()
```

Comment:
If `country_code` and `user_id` don't change during the lifetime of a `Tidal` object, maybe it would be better to have them as private immutable fields of the `Tidal` struct initialized only on `::new()`?

Reply:
@jpalvarezl for me `country_code` and `user_id` are properties of the `Credential` (maybe merge `Credential` with `SessionInfo` might be a better idea, I'm not sure), I am not sure they should be a property of the `Tidal` library itself.

Ideally in the future the library could allow to use multiple credencials `tidal_client.set_credential(credential); // considering tidal_client is an"instance of Tidal`, for applications to allow to change user or to use a persisted session.

Maybe this convenience methods to access the session properties should not exist 🤷‍♂️

Reply:
I created an issue for this: https://github.com/4xposed/rstidal/issues/2